iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -555,6 +499,34 @@ struct SubTensorOpConversion
 };
 }  // namespace
 
+namespace {
+/// Converts subtensor_insert operation to subview + linalg.copy
+struct SubTensorInsertOpConversion
+    : public ConvertToLinalgBufferOp<SubTensorInsertOpConversion,
+                                     SubTensorInsertOp> {
+  using ConvertToLinalgBufferOp<SubTensorInsertOpConversion,
+                                SubTensorInsertOp>::ConvertToLinalgBufferOp;
+
+  LogicalResult apply(SubTensorInsertOp op, ArrayRef<Value> inputBuffers,
+                      ArrayRef<Value> resultBuffers,
+                      ConversionPatternRewriter &rewriter) const {
+    auto loc = op.getLoc();
+    auto subViewOp =
+        rewriter.create<SubViewOp>(loc, resultBuffers[0], op.getMixedOffsets(),
+                                   op.getMixedSizes(), op.getMixedStrides());
+    if (auto cstOp = inputBuffers[0].getDefiningOp<ConstantOp>()) {
```

Comment:
It might not always be a splat constant.

Reply:
This is keeping the same behavior as before. I know in the near future we want to inline small constants. I think in this case, it will be passed with a `tensor_to_memref`, and we can do a linalg.copy here.

For now, I cast it to splat constants, so if something goes wrong, it will complain.

Reply:
Oh, I think a better way is to return notifyMatchFailure. Let me update it.

Reply:
Have been looking into supporting subtensor_insert op on the linalg on tensors path (see WIP commit here https://github.com/MaheshRavishankar/iree/commit/1897ebb7e390bd1fb7ee8f325c965f0df02d9c19). The subtensor_insert does not lower directly to a subview + copy. It actually lowers to two copies.

```%c_tensor = subtensor_insert %a_tensor into %b_tensor [...]...```

becomes
```
linalg.copy(%b_buffer, %c_buffer)
%subview = subview %c_buffer[...]...
linalg.copy(%a_buffer, %subview)
```

Now these `copy` operations need to be put in different dispatches since they have different workload and need cross workgroup synchronization for correctness. Trying to understand why this works.

Reply:
Spoke to Hanhan offline. Seems like the pre-analysis sets the dest and result to the same buffer so that everything happens in place. Still might be better to do a copy from dest -> result if they dont happen to be the same.  This path is probably going away soon. So shouldnt matter

Reply:
yeah, the path is probably going away soon, I think we won't modify the dispatch function creation logic in the old path anymore, so it shouldn't matter and let's keep it simple. I will add a comment for it.